data/sql/updates/pending_db_characters/rev_1624468542752528700.sql

```diff
@@ -0,0 +1,19 @@
+INSERT INTO `version_db_characters` (`sql_rev`) VALUES ('1624468542752528700');
+
+SET @dbname = DATABASE();
+SET @tablename = "characters";
+SET @columnname = "order";
+SET @preparedStatement = (SELECT IF(
+    (
+        SELECT COUNT(*) FROM information_schema.COLUMNS
+        WHERE
+            (TABLE_NAME = @tablename)
+            AND (TABLE_SCHEMA = @dbname)
+            AND (COLUMN_NAME = @columnname)
+    ) > 0,
+    "SELECT 1 WHERE false",
+    CONCAT("ALTER TABLE `", @tablename, "` ADD `", @columnname, "` TINYINT NULL DEFAULT NULL AFTER `grantableLevels`;")
+));
+PREPARE alterIfNotExists FROM @preparedStatement;
+EXECUTE alterIfNotExists;
+DEALLOCATE PREPARE alterIfNotExists;
```

Comment:
What does this do over

```sql
ALTER TABLE characters
  ADD order TINYINY NULL DEFAULT NULL AFTER granableLevels

Reply:
It ensures we can run it multiple times without throwing an error, it's the column version of "always do a DELETE query before an INSERT"

Reply:
As you can see in the commit history, at first I did
```sql
ALTER TABLE `characters` DROP IF EXISTS `order`;
```
before the `ADD`, and it worked well, but the `IF EXISTS` part only works in mariadb, this doesn't exist in mysql, so I had to resort to this

Reply:
It does exist, or is it different? https://dev.mysql.com/doc/refman/8.0/en/drop-table.html

Reply:
Maybe it is only for tables and not columns

Reply:
nvm. It's only for DROP and not ALTER https://bugs.mysql.com/bug.php?id=10789

Reply:
Yes, this is only for tables. There's a [feature request for columns from 2005](https://bugs.mysql.com/bug.php?id=10789), still nothing 😒

Reply:
Oh you linked the same thing as I was typing it haha